samples/spring-boot-auto-config/src/test/resources/application.yaml

```diff
@@ -0,0 +1,8 @@
+io.javaoperatorsdk:
```

Comment:
I don't think this is a good way of defining config tree. 
I would go for `javaoperatorsdk.*` and `javaoperatorsdk.test.*`

Reply:
I completely agree, let's change the naming in another PR as it will affect other parts too.